diff --git a/conf.d/__async_prompt.fish b/conf.d/__async_prompt.fish
index d7fcb5e..1234567 100644
--- a/conf.d/__async_prompt.fish
+++ b/conf.d/__async_prompt.fish
@@ -15,6 +15,7 @@ function __async_prompt_setup

     for func in (__async_prompt_config_functions)
         function $func -V func
+            contains -- --final-rendering $argv; and return
             test -e $__async_prompt_tmpdir'/'$fish_pid'_'$func
             and cat $__async_prompt_tmpdir'/'$fish_pid'_'$func
         end
