--- a/conf.d/__async_prompt.fish
+++ b/conf.d/__async_prompt.fish
@@ -57,6 +57,10 @@ function __async_prompt_setup_on_startup --on-event fish_prompt
         # Replace the fish_prompt functions with replacements that cat the
         # sync and async prompts from the tmpdir
         function $func -V func
+            if test (count $argv) -gt 0
+                __async_prompt_orig_$func $argv
+                return
+            end
             test -e $__async_prompt_tmpdir'/'$fish_pid'_'$func
             # Using `string collect` as a faster cat.
             and string collect <$__async_prompt_tmpdir'/'$fish_pid'_'$func
