diff --git a/cps/server.py b/cps/server.py
index fb70a723..39da4e9e 100644
--- a/cps/server.py
+++ b/cps/server.py
@@ -230,7 +230,7 @@ class WebServer(object):
 
         # Max Buffersize set to 200MB
         http_server = HTTPServer(MyWSGIContainer(self.app),
-                                 max_buffer_size=209700000,
+                                 max_buffer_size=2048000000,
                                  ssl_options=self.ssl_args)
         http_server.listen(self.listen_port, self.listen_address)
         self.wsgiserver = IOLoop.current()
